### PR TITLE
Minor Build Fixes For MacOS

### PR DESCRIPTION
**Pull request by CuriousTommy**
Just some minor build issues I noticed when building for macOS.

### CODE REVIEW

File: include/blobify/load.hpp [hunk 1]
```diff
@@ -13,6 +13,7 @@
 #include <magic_enum.hpp>
 
 #include <cstddef>
+#include <vector>
```
Comment (neobrain) [inline include/blobify/load.hpp]: What definition from `<vector>` is needed by load.hpp? I can't see any use of `std::vector` here.

If only some of the files that include `load.hpp` require `<vector>`'s definitions, would you mind including the header in those files instead?

File: include/blobify/endian.hpp [hunk 2]
```diff
@@ -5,8 +5,12 @@
 // C++20 provides std::endian
 #include <bit>
 #elif !defined(_WIN32)
+#ifdef __APPLE__
```
Comment (neobrain) [inline include/blobify/endian.hpp]: Mind folding this ifdef into the outer one? I.e.

```cpp
#if defined(__cpp_lib_endian)
...
#elif defined(__APPLE__)
...
#elif !defined(_WIN32)
...
#endif
```

**Comment by neobrain** [comment]
Hey, thanks for the build fix! Sorry for my late review, it turns out notifications for this repository weren't working for me.

File: include/blobify/load.hpp (hunk 1 above)
Comment (CuriousTommy) [reply include/blobify/load.hpp]: I just realized that `examples/bmp.cpp` was were `#include <vector>` should actually belong.

File: include/blobify/endian.hpp (hunk 2 above)
Comment (CuriousTommy) [reply include/blobify/endian.hpp]: Fixed

**Comment by CuriousTommy** [comment]
No problem! Let me know if you need me to do any other changes.

**Comment by neobrain** [comment]
Perfect, thanks! I just manually merged this via 22af9d93df7798b4e30eed34d1a108a5eabcd81c to keep the git history clean. Thanks for your help!

**Pull request closed without merging**